services/src/main/java/org/keycloak/services/clientpolicy/condition/AutomaticClientRegistrationCondition.java

```diff
@@ -0,0 +1,133 @@
+import org.keycloak.utils.OpenIdFederationUtils;
+import org.keycloak.crypto.SignatureProvider;
+import org.keycloak.jose.jwk.JWK;
+import org.keycloak.util.JWKSUtils;
+import org.keycloak.OAuthErrorException;
+import org.keycloak.crypto.PublicKeysWrapper;
+
+
+public class AutomaticClientRegistrationCondition extends AbstractClientPolicyConditionProvider<ClientPolicyConditionConfigurationRepresentation> {
+
+    private static final Logger logger = Logger.getLogger(AutomaticClientRegistrationCondition.class);
+    private final TrustChainProcessor trustChainProcessor;
+    
+    public AutomaticClientRegistrationCondition(KeycloakSession session) {
+        super(session);
+        this.trustChainProcessor = session.getProvider(TrustChainProcessor.class, OpenIdFederationTrustChainProcessorFactory.PROVIDER_ID);
+    }
+
+    @Override
+    public Class<ClientPolicyConditionConfigurationRepresentation> getConditionConfigurationClass() {
+        return ClientPolicyConditionConfigurationRepresentation.class;
+    }
+
+    @Override
+    public String getProviderId() {
+        return AutomaticClientRegistrationConditionFactory.PROVIDER_ID;
+    }
+
+    @Override
+    public ClientPolicyVote applyPolicy(ClientPolicyContext context) throws ClientPolicyException {      
+
+        switch (context.getEvent()) {
+            case PRE_AUTHORIZATION_REQUEST:
+                
+                RealmModel realm = session.getContext().getRealm();
+                PreAuthorizationRequestContext paContext = (PreAuthorizationRequestContext) context;
+                ClientModel client = session.getContext().getRealm().getClientByClientId(paContext.getClientId());
+                String requestObject = paContext.getRequestParameters().getFirst(OIDCLoginProtocol.REQUEST_PARAM);
+
+                if (client == null) {
+                    if (realm.isOpenIdFederationTypeRegistrationSupported(EntityTypeEnum.OPENID_PROVIDER, ClientRegistrationTypeEnum.AUTOMATIC) && !realm.getOpenIdFederations().isEmpty()) {
+
+                        try {  
+                            // get Entity Configuration from RP's well-known endpoint
+                            String entityConfigurationOfRP = OpenIdFederationUtils.getSelfSignedToken(paContext.getClientId(), session);
+                            
+                            // validate the Configuration and get Entity Statement object which is used to get JWKS of the RP
+                            EntityStatement entityStatementFromRPConfiguraionEndpoint = trustChainProcessor.parseAndValidateSelfSigned(entityConfigurationOfRP);
+      
+                            // validate the request object with the JWKS of RP which is got from the well-known endpoint
+                            if (requestObject == null) {
+                                throw new ClientPolicyException(OAuthErrorException.INVALID_REQUEST_OBJECT, "request object is null.");
+                            }
```

Comment:
Request object is only one way to authorize even with automatic client registration in OpenId Federation.
Maybe mentioned as unimplemented